src/e2e-test/java/io/cdap/plugin/pubsub/stepsdesign/PubSubConnector.java

```diff
@@ -0,0 +1,401 @@
+    }
+  }
+
+  @Then("Verify Preview output schema matches the outputSchema captured in properties")
+  public void verifyPreviewOutputSchemaMatchesTheOutputSchemaCapturedInProperties() {
+    List<String> previewOutputSchema = new ArrayList<String>();
+    List<WebElement> previewOutputSchemaElements = SeleniumDriver.getDriver().findElements(
+      By.xpath("(//h2[text()='Output Records']/parent::div/div/div/div/div)[1]//div[text()!='']"));
+    for (WebElement element : previewOutputSchemaElements) {
+      previewOutputSchema.add(element.getAttribute("title"));
+    }
+    Assert.assertTrue(previewOutputSchema.equals(propertiesOutputSchema));
+  }
+
+  @Then("Close the Preview")
+  public void closeThePreview() {
+    PubSubLocators.closeButton.click();
+    CdfStudioActions.previewSelect();
+  }
+
+  @Then("Deploy the pipeline")
+  public void deployThePipeline() {
+    SeleniumHelper.waitElementIsVisible(CdfStudioLocators.pipelineDeploy, 20);
+    CdfStudioActions.pipelineDeploy();
+  }
+
+  @Then("Validate successMessage is displayed")
+  public void validateSuccessMessageIsDisplayed() {
+    CdfLogActions.validateSucceeded();
+  }
+
+  @Then("Click on PreviewData for GCS Connector")
+  public void clickOnPreviewDataForGCSConnector() {
+    CdfGcsActions.clickPreviewData();
+  }
+
+  @When("Source is BigQuery")
+  public void sourceIsBigQuery() throws InterruptedException {
+    CdfStudioActions.selectBQ();
+  }
+
+  @Then("Open BigQuery Properties")
+  public void openBigQueryProperties() {
+    CdfStudioActions.clickProperties("BigQuery");
+  }
+
+  @Then("Enter the BigQuery properties for table {string}")
+  public void enterTheBigQueryPropertiesForTable(String tableName) throws IOException {
+    CdfGcsActions.enterProjectId();
+    CdfBigQueryPropertiesActions.enterProjectId(E2ETestUtils.pluginProp("projectId"));
+    CdfBigQueryPropertiesActions.enterDatasetProjectId(E2ETestUtils.pluginProp("projectId"));
+    CdfBigQueryPropertiesActions.enterBigQueryReferenceName("BQ_Ref_" + UUID.randomUUID().toString());
+    CdfBigQueryPropertiesActions.enterBigQueryDataset(E2ETestUtils.pluginProp("dataset"));
+    CdfBigQueryPropertiesActions.enterBigQueryTable(E2ETestUtils.pluginProp(tableName));
+  }
+
+  @Then("Close the BigQuery properties")
+  public void closeTheBigQueryProperties() {
+    CdfStudioActions.clickCloseButton();
+  }
+
+  @Then("Click on PreviewData for BigQuery Connector")
+  public void clickOnPreviewDataForBigQueryConnector() {
+    CdfBigQueryPropertiesActions.clickPreviewData();
+  }
+
+  @Then("Get Count of no of records transferred to BigQuery in {string}")
+  public void getCountOfNoOfRecordsTransferredToBigQueryIn(String tableName) throws IOException, InterruptedException {
+    int countRecords = gcpClient.countBqQuery(E2ETestUtils.pluginProp(tableName));
+    BeforeActions.scenario.write("**********No of Records Transferred******************:" + countRecords);
+    Assert.assertTrue(countRecords > 0);
```

Comment:
verify actual number of records written, not just > 0

Reply:
changed

Reply:
where did you fix it? I don't see this method in the latest code

Reply:
Added new step to verify "out" record count is equal to "In" record count